cmd/nginx-ingress/flags.go

```diff
@@ -203,8 +201,9 @@ var (
 
 	enableDynamicSSLReload = flag.Bool(dynamicSSLReloadParam, true, "Enable reloading of SSL Certificates without restarting the NGINX process.")
 
-	enableTelemetryReporting = flag.Bool("enable-telemetry-reporting", true, "Enable gathering and reporting of product related telemetry.")
-	telemetryReportingPeriod = flag.String("telemetry-reporting-period", "24h", "Sets a telemetry reporting period.")
+	enableTelemetryReporting   = flag.Bool("enable-telemetry-reporting", true, "Enable gathering and reporting of product related telemetry.")
+	telemetryReportingEndpoint = flag.String("telemetry-reporting-endpoint", "dev-oss.edge.df.f5.com:443", "Set the GRPC endpoint to send data to.")
```

Comment:
@shaun-nx, general question: Could you help me to understand: Why does NIC allow users to change telemetry endpoint? The same question about having `insecure` endpoint option in NIC.

Reply:
@jjngx Having this be configurable is useful for both development and in testing.
For testing, if we want to send data to a local otel collector, we may need to set. the `secure` option to false. I've had to do this when I was testing locally and switching between my local collector and the dev XCDF endpoint.

That being said, it may not make sense to present this as a configuration option here, and instead set it as a build option. I know that NFG did it this way too.

Reply:
Let's follow the same route. For local testing we do not need even build params. We can change it directly in the code and then remove (before sending a PR). This would be the same approach as with the reporting period. Users should not be given an option to alter these params.

Reply:
What do python tests actually test and prove in this particular case? Allowing users to change the telemetry endpoint does not make any sense from business perspective. What other product on the market, that gathers usage stats, allows end-users to mess with the endpoint?

My2c